pkg/alt/karpenter-core/pkg/operator/operator.go

```diff
@@ -202,15 +182,49 @@ func NewOperator() (context.Context, *coreoperator.Operator) {
 	mgr, err := ctrl.NewManager(overlayConfig, mgrOpts)
 	mgr = lo.Must(mgr, err, "failed to setup manager")
 
-	lo.Must0(mgr.GetFieldIndexer().IndexField(overlayCtx, &corev1.Pod{}, "spec.nodeName", func(o client.Object) []string {
+	setupIndexers(overlayCtx, ccPlaneCtx, mgr)
+
+	lo.Must0(mgr.AddReadyzCheck("manager", func(req *http.Request) error {
+		return lo.Ternary(mgr.GetCache().WaitForCacheSync(req.Context()), nil, fmt.Errorf("failed to sync caches"))
+	}))
+	lo.Must0(mgr.AddReadyzCheck("crd", func(_ *http.Request) error {
+		objects := []client.Object{&v1.NodePool{}, &v1.NodeClaim{}}
+		for _, obj := range objects {
+			gvk, err := apiutil.GVKForObject(obj, scheme.Scheme)
+			if err != nil {
+				return err
+			}
+			if _, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
+				return err
+			}
+		}
+		return nil
+	}))
+	lo.Must0(mgr.AddHealthzCheck("healthz", healthz.Ping))
+	lo.Must0(mgr.AddReadyzCheck("readyz", healthz.Ping))
+
+	return ccPlaneCtx, &coreoperator.Operator{
+		Manager:             mgr,
+		KubernetesInterface: overlayKubernetesInterface,
+		EventRecorder:       events.NewRecorder(mgr.GetEventRecorderFor(appName)),
+		Clock:               clock.RealClock{},
+	}
+}
+
+func setupIndexers(ctx context.Context, ccPlaneCtx context.Context, mgr manager.Manager) {
```

Comment:
I dont think the `ccPlaneCtx` is used here.

Reply:
It is used, for the logger - but I am starting to think may be we now don't need the alternative operator at all